src/components/map/MapPage.js

```diff
@@ -87,18 +51,43 @@ const MapPage = () => {
   const [view, setView] = useState(DEFAULT_VIEW_STATE)
   const [locations, setLocations] = useState([])
   const [clusters, setClusters] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
 
   useEffect(() => {
-    if (view.zoom <= VISIBLE_CLUSTER_ZOOM_LIMIT) {
-      // TODO: Fetch cluster data from server
-      setClusters(clusterData)
-      setLocations([])
-    } else {
-      // TODO: Fetch location data from server
-      setLocations(locationData)
-      setClusters([])
+    async function fetchClusterAndLocationData() {
+      if (view.bounds) {
+        setIsLoading(true)
+        if (view.zoom <= VISIBLE_CLUSTER_ZOOM_LIMIT) {
+          const query = {
+            swlng: view.bounds.sw.lng,
+            nelng: view.bounds.ne.lng,
+            swlat: view.bounds.sw.lat,
+            nelat: view.bounds.ne.lat,
+            zoom: view.zoom,
+            muni: 1,
+          }
+          const clusters = await getClusters(query)
+          setClusters(clusters)
+          setLocations([])
+        } else {
+          const query = {
+            swlng: view.bounds.sw.lng,
+            nelng: view.bounds.ne.lng,
+            swlat: view.bounds.sw.lat,
+            nelat: view.bounds.ne.lat,
+            muni: 1,
+          }
+          const locations = await getLocations(query)
+          // Remove number of locations returned and total locations available from result
+          locations.splice(0, 2)
```

Comment:
Best to write this as `const [numLocations, totalLocations, ...locations] = await getLocations(query)`